Source/Scene/Model.js

```diff
@@ -4094,20 +4018,32 @@ define([
                 }
 
                 this._loadResources = new LoadResources();
-                parse(this);
             }
+            parse(this);
         }
 
         var loadResources = this._loadResources;
         var incrementallyLoadTextures = this._incrementallyLoadTextures;
         var justLoaded = false;
 
         if (this._state === ModelState.LOADING) {
-            // Create WebGL resources as buffers/shaders/textures are downloaded
-            createResources(this, frameState);
-
             // Transition from LOADING -> LOADED once resources are downloaded and created.
             // Textures may continue to stream in while in the LOADED state.
+            if (loadResources.pendingBufferLoads === 0 && loadResources.pendingShaderLoads === 0) {
+                if (!this._updatedGltfVersion) {
+                    updateVersion(this.gltf);
```

Comment:
> What is the issue with the previous approach and storing things in _loadResources? We're using an implementation detail of gltf-pipeline to store data in those extras properties. Are you sure that the glTF objects need to be augmented at all instead of looking things by id in _loadResources?

Things are still stored in _loadResources. We are just adding a reference to the same data in the place where the gltf-pipeline expects it to be. I am adding `removePipelineExtras` from the pipeline, so this should be a non-issue.